launcher/modManager/cmodlistview_moc.cpp

```diff
@@ -1043,12 +1083,91 @@ void CModListView::installMods(QStringList archives)
 void CModListView::installMaps(QStringList maps)
 {
 	const auto destDir = CLauncherDirs::mapsPath() + QChar{'/'};
+	int successCount = 0;
+	QStringList failedMaps;
 
 	for(QString map : maps)
 	{
+		// ZIP archive handling
+		if (map.endsWith(".zip", Qt::CaseInsensitive))
+		{
+			logGlobal->info("Importing map archive '%s'", map.toStdString());
+
+			ZipArchive archive(qstringToPath(map));
+			auto fileList = archive.listFiles();
+
+			for (const auto& file : fileList)
+			{
+				QString name = QString::fromStdString(file);
+
+				if (!(name.endsWith(".h3m", Qt::CaseInsensitive) || name.endsWith(".h3c", Qt::CaseInsensitive) || name.endsWith(".vmap", Qt::CaseInsensitive) || name.endsWith(".vcmp", Qt::CaseInsensitive)))
+					continue;
+
+				QString destFile = destDir + name;
+
+				// prompt if exists
+				if (QFile::exists(destFile))
+				{
+					int ret = QMessageBox::question(this, tr("Map exists"), tr("Map '%1' already exists. Do you want to overwrite it?").arg(name), QMessageBox::Yes | QMessageBox::No, QMessageBox::No);
```

Comment:
Can we also add "yes to all" / "no to all" to this dialog &  handle it?

Othervice importing large map pack for 2nd time by accident might be painful

Reply:
But it's called for each map standalone. So when you select 2 h3ms, you will never reach "Yes to all" as it seems when you select more maps they are not placed into list. It's probably bug, in import dialog processing. In any case I have added it.